architecture/new_schema.sql

```diff
@@ -0,0 +1,1060 @@
+
+ALTER TABLE public.operation OWNER TO veramo;
+
+--
+-- Name: payment; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public.payment (
+    "txHash" character varying NOT NULL,
+    "customerId" uuid NOT NULL,
+    direction public.paymentdirectionenum NOT NULL,
+    fee bigint,
+    "identifierDid" character varying,
+    "resourceId" uuid,
+    "paymentAddress" character varying NOT NULL,
+    "operationId" uuid NOT NULL,
+    "timestamp" date
+);
+
+
+ALTER TABLE public.payment OWNER TO veramo;
+
+--
+-- Name: paymentAccount; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public."paymentAccount" (
+    namespace public.namespaceenum DEFAULT 'testnet'::public.namespaceenum NOT NULL,
+    "customerId" uuid NOT NULL,
+    kid character varying NOT NULL,
+    "createdAt" date NOT NULL,
+    address character varying NOT NULL,
+    "isDefault" boolean DEFAULT false NOT NULL,
+    "updatedAt" date
+);
+
+
+ALTER TABLE public."paymentAccount" OWNER TO veramo;
+
+--
+-- Name: presentation; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public.presentation (
+    hash character varying NOT NULL,
+    raw text NOT NULL,
+    id character varying,
+    "issuanceDate" timestamp without time zone,
+    "expirationDate" timestamp without time zone,
+    context text NOT NULL,
+    type text NOT NULL,
+    "holderDid" character varying,
+    "customerId" uuid
+);
+
+
+ALTER TABLE public.presentation OWNER TO veramo;
+
+--
+-- Name: presentation_credentials_credential; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public.presentation_credentials_credential (
+    "presentationHash" character varying NOT NULL,
+    "credentialHash" character varying NOT NULL
+);
+
+
+ALTER TABLE public.presentation_credentials_credential OWNER TO veramo;
+
+--
+-- Name: presentation_verifier_identifier; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public.presentation_verifier_identifier (
+    "presentationHash" character varying NOT NULL,
+    "identifierDid" character varying NOT NULL
+);
+
+
+ALTER TABLE public.presentation_verifier_identifier OWNER TO veramo;
+
+--
+-- Name: private-key; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public."private-key" (
+    alias character varying NOT NULL,
+    type character varying NOT NULL,
+    "privateKeyHex" character varying NOT NULL
+);
+
+
+ALTER TABLE public."private-key" OWNER TO veramo;
+
+--
+-- Name: resource; Type: TABLE; Schema: public; Owner: veramo
+--
+
+CREATE TABLE public.resource (
+    "resourceId" uuid NOT NULL,
+    "identifierDid" character varying NOT NULL,
+    kid character varying NOT NULL,
+    "resourceName" character varying NOT NULL,
+    "resourceType" character varying,
+    "mediaType" character varying NOT NULL,
+    "previousVersionId" uuid,
+    "createdAt" date NOT NULL,
+    "nextVersionId" uuid,
+    "customerId" uuid NOT NULL,
+    "updatedAt" date,
+    "symmetricKey" character varying
+);
```

Comment:
Maybe add `encrypted` as boolean true/false? This could denote whether the resource is encrypted or not. Since the `resourceType` on its own does not tell us if the resource is encrypted or not.

Reply:
I can add the explicit field "encrypted" but I was thinking about using `symmetricKey` field as a marker. If it's empty - resource is unencrypted and vice verse

Reply:
I added `encrypted` field for `resources` into diagrams, images and docs